DllImportGenerator/DllImportGenerator.IntegrationTests/BooleanTests.cs

```diff
@@ -0,0 +1,113 @@
+
+        [GeneratedDllImport(nameof(NativeExportsNE), EntryPoint = "bool_return_as_uint")]
+        public static partial uint ReturnIntBoolAsUInt([MarshalAs(UnmanagedType.I4)] bool input);
+
+        [GeneratedDllImport(nameof(NativeExportsNE), EntryPoint = "bool_return_as_uint")]
+        public static partial uint ReturnUIntBoolAsUInt([MarshalAs(UnmanagedType.U4)] bool input);
+
+        [GeneratedDllImport(nameof(NativeExportsNE), EntryPoint = "bool_return_as_uint")]
+        public static partial uint ReturnWinBoolAsUInt([MarshalAs(UnmanagedType.Bool)] bool input);
+
+        [GeneratedDllImport(nameof(NativeExportsNE), EntryPoint = "bool_return_as_uint")]
+        [return: MarshalAs(UnmanagedType.U1)]
+        public static partial bool ReturnUIntAsByteBool(uint input);
+
+        [GeneratedDllImport(nameof(NativeExportsNE), EntryPoint = "bool_return_as_uint")]
+        [return: MarshalAs(UnmanagedType.VariantBool)]
+        public static partial bool ReturnUIntAsVariantBool(uint input);
+
+        [GeneratedDllImport(nameof(NativeExportsNE), EntryPoint = "bool_return_as_uint")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static partial bool ReturnUIntAsWinBool(uint input);
+    }
+
+    public class BooleanTests
+    {
+        // See definition of Windows' VARIANT_BOOL
+        const ushort VARIANT_TRUE = unchecked((ushort)-1);
+        const ushort VARIANT_FALSE = 0;
+
+        [Fact]
+        public void ValidateBoolIsMarshalledAsExpected()
+        {
+            Assert.Equal((uint)1, NativeExportsNE.ReturnByteBoolAsUInt(true));
+            Assert.Equal((uint)0, NativeExportsNE.ReturnByteBoolAsUInt(false));
+            Assert.Equal((uint)1, NativeExportsNE.ReturnSByteBoolAsUInt(true));
+            Assert.Equal((uint)0, NativeExportsNE.ReturnSByteBoolAsUInt(false));
+            Assert.Equal(VARIANT_TRUE, NativeExportsNE.ReturnVariantBoolAsUInt(true));
+            Assert.Equal(VARIANT_FALSE, NativeExportsNE.ReturnVariantBoolAsUInt(false));
+            Assert.Equal((uint)1, NativeExportsNE.ReturnIntBoolAsUInt(true));
+            Assert.Equal((uint)0, NativeExportsNE.ReturnIntBoolAsUInt(false));
+            Assert.Equal((uint)1, NativeExportsNE.ReturnUIntBoolAsUInt(true));
+            Assert.Equal((uint)0, NativeExportsNE.ReturnUIntBoolAsUInt(false));
+            Assert.Equal((uint)1, NativeExportsNE.ReturnWinBoolAsUInt(true));
+            Assert.Equal((uint)0, NativeExportsNE.ReturnWinBoolAsUInt(false));
+        }
+
+        public static IEnumerable<object[]> ByteBoolReturns()
+        {
+            yield return new object[] { 0, false };
+            yield return new object[] { 1, true };
+            yield return new object[] { 37, true };
+            yield return new object[] { 0xff, true };
+            yield return new object[] { 0xffffff00, false };
+        }
+
+        [Theory]
+        [MemberData(nameof(ByteBoolReturns))]
```

Comment:
Nit: I'd prefer `InlineData` since the data is not shared with other tests and there aren't that many cases